src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectDefaultConverter.cs

```diff
@@ -28,7 +28,7 @@ internal override bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert,
 
                 if (state.Current.JsonClassInfo.CreateObject == null)
                 {
-                    ThrowHelper.ThrowNotSupportedException_DeserializeNoDeserializationConstructor(state.Current.JsonClassInfo.Type, ref reader, ref state);
+                    ThrowHelper.ThrowNotSupportedException_DeserializeNoConstructor(state.Current.JsonClassInfo.Type, ref reader, ref state);
                 }
 
                 obj = state.Current.JsonClassInfo.CreateObject!()!;
```

Comment:
Is comment on line 64 accurate? As per my understanding, we don't support custom ctors when `ReferenceHandling.Preserve` is on.

Reply:
This isn't the converter for objects with parameterized ctors. That's in `ObjectWithParameterizedConstructorConverter`. The exception for metadata being in the payload when deserializing with parameterized ctors + `Preserve` is thrown here: https://github.com/dotnet/runtime/blob/960ea5a9eac90c28416cb32dc9797ee5c3ebc5f4/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs#L115.

Reply:
Oh sorry, I was thinking about this line: https://github.com/dotnet/runtime/blob/847a054cfd86f8ea828003197fda644a2fb20dbb/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs#L75

Reply:
Ah, yup. I've edited the comment. Thanks.